fast_llm/tools/prepare_dataset.py

```diff
@@ -0,0 +1,359 @@
+import abc
+import argparse
+import json
+import os
+import pathlib
+import shutil
+import typing
+from multiprocessing import Pool
+
+import numpy as np
+import torch.distributed
+
+from fast_llm.config import Config, Field, FieldHint, check_field, config_class
+from fast_llm.data.config import TokenizerConfig
+from fast_llm.data.gpt.memmap import GPTMemmapDataset
+from fast_llm.data.tokenizer import Tokenizer
+from fast_llm.engine.config_utils.data_type import DataType
+from fast_llm.engine.config_utils.runnable import RunnableConfig
+from fast_llm.utils import Assert, Registry
+
+
+@config_class
+class DistributedConfig(Config):
```

Comment:
This is basically a duplicate of the other distributed config, are you really sure we can't merge them? What would be the minimal requirement requirement for combining them?

Reply:
See https://github.com/ServiceNow/Fast-LLM/pull/38#issuecomment-2468184561
I need help in unifying these and remove any hard dependency on GPUs and CUDA from what we have in Fast-LLM

Reply:
Which hard dependencies? From what I see there is already a CPU-only option, but it comes with an `assert world_size == 1`. Apart from that, I think only the default nccl backend is a problem. Is there anything else?

For now, a compromise would be to use `DistributedConfig` but maybe not `Distributed`. The default backend could be changed with a FieldUpdate in a derived class, and ignoring a bunch of config parameters. What do you think?

Reply:
Let's leave this for another PR please.